src/codewhisperer/activation.ts

```diff
@@ -47,6 +47,11 @@ import { RecommendationHandler } from './service/recommendationHandler'
 const performance = globalThis.performance ?? require('perf_hooks').performance
 
 export async function activate(context: ExtContext): Promise<void> {
+    const codewhispererSettings = CodeWhispererSettings.instance
+    if (!codewhispererSettings.isEnabled()) {
+        return
+    }
```

Comment:
Right now the user has to reload the extension which is fine as long as they know that's what they have to do.

You can also do it dynamically with something like this, wrapping the existing activation function:
```ts
const activateModule = once(async (context: ExtContext) => {
    try {
        await activate(context)
    } catch (err) {
        getLogger().error(`modules: failed to activate CodeWhisperer: %O`, err)
        showViewLogsMessage('Failed to enable CodeWhisperer')
    }
})

const experiments = Experiments.instance
if (await experiments.isExperimentEnabled('CodeWhisperer')) {
    return activateModule()
} else {
    experiments.onDidChange(({ key }) => {
        if (key === 'CodeWhisperer') {
            activateModule()
        }
    })
}
```

Reply:
That is good to know. I intentionally wanted to keep it simple to avoid regressions, and because this is only a troubleshooting step, so "restart vscode" isn't too bad of an extra step then.